### DISCUSSION TITLE
What is the closest way to trim the end of the stream returned at a given time, while staying on client side (without using ffmpeg or web assembly solutions)

### DISCUSSION THREAD

**Discussion by mgloc** [fent/node-ytdl-core#1061] [Q&A]
Indeed slicing blob end doesn't affect the head keyframe, and then the downloaded resulting file would have incorrect duration

**Comment by TheNozomi** [comment]
As you can check on [this SO answer](https://stackoverflow.com/a/46609643), you can't simply remove bytes from the video to trim it, you also have to modify the atom headers (that's for mp4, matroska uses a similar structure). Since you don't want to use FFMpeg I guess you're trying to do this process client-side (browser), if so, any reason for not wanting to use wasm?
